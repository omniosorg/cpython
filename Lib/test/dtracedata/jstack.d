
python$target:::function-entry
/copyinstr(arg1)=="test_stack"/
{
    self->trace = 1;
}
python$target:::function-entry
/self->trace/
{
    printf("[x]");
    jstack();
}
python$target:::function-return
/copyinstr(arg1)=="test_stack"/
{
    self->trace = 0;
}

