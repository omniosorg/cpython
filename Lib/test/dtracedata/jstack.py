
def function_1():
    pass

def function_2():
    function_1()

def function_3(dummy, dummy2):
    pass

def function_4(**dummy):
    pass

def function_5(dummy, dummy2, **dummy3):
    pass

def test_stack():
    function_1()
    function_2()
    function_3(*(1,2))
    function_4(**{"test":42})
    function_5(*(1,2), **{"test":42})

test_stack()

