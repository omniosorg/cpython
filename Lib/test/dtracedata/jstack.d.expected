[x]
[PyFile:17(test_stack)]
[x]
[PyFile:2(function_1)]
[PyFile:17(test_stack)]
[x]
[PyFile:5(function_2)]
[PyFile:17(test_stack)]
[x]
[PyFile:2(function_1)]
[PyFile:5(function_2)]
[PyFile:17(test_stack)]
[x]
[PyFile:8(function_3)]
[PyFile:18(test_stack)]
[x]
[PyFile:11(function_4)]
[PyFile:19(test_stack)]
[x]
[PyFile:14(function_5)]
[PyFile:20(test_stack)]